mlir/test/Dialect/Vector/vector-interleave-to-shuffle.mlir

```diff
@@ -0,0 +1,21 @@
+// RUN: mlir-opt %s --transform-interpreter | FileCheck %s
+
+// CHECK-LABEL: @vector_interleave_to_shuffle
+func.func @vector_interleave_to_shuffle(%a: vector<7xi16>, %b: vector<7xi16>) -> vector<14xi16>
+{
+  %0 = vector.interleave %a, %b : vector<7xi16>
+  return %0 : vector<14xi16>
+}
+// CHECK: vector.shuffle %arg0, %arg1 [0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6] : vector<7xi16>, vector<7xi16>
```

Comment:
Actually, I'm not sure about this one anymore. The `vector.interleave` op docs don't really define what it does beyond saying (`interleave` interleaves).

But for `shuffle`, the indices are over a temporary vector of concatenated operands. Assuming a reasonable interpretation, I'd expect `0, 7, 1, 8, 2, 9, ...` here.

Reply:
As for shuffle, the llvm and spir-v docs are pretty clear about indexing:
* https://llvm.org/docs/LangRef.html#shufflevector-instruction
* https://mlir.llvm.org/docs/Dialects/SPIR-V/#spirvvectorshuffle-spirvvectorshuffleop

Reply:
Thanks for catching that. Pushed a fix.